backend/src/main/java/ch/puzzle/okr/multitenancy/TenantConfigProvider.java

```diff
@@ -1,56 +1,120 @@
 package ch.puzzle.okr.multitenancy;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.core.env.Environment;
 import org.springframework.stereotype.Component;
 
 import java.text.MessageFormat;
 import java.util.*;
 
+import static java.text.MessageFormat.format;
+
+/**
+ * Reads the configuration of the tenants (as TenantConfig objects) from the applicationX.properties and caches each
+ * TenantConfig in the TenantConfigs class.
+ */
 @Component
 public class TenantConfigProvider implements TenantConfigProviderInterface {
     private static final String EMAIL_DELIMITER = ",";
     private final Map<String, TenantConfig> tenantConfigs = new HashMap<>();
     private final Environment env;
 
+    private enum DbType {
+        BOOTSTRAP, APP, FLY;
+
+        public String nameUsedInProperties() {
+            return this.name().toLowerCase();
+        }
+    }
+
+    private static final Logger logger = LoggerFactory.getLogger(TenantConfigProvider.class);
+
     public TenantConfigProvider(final @Value("${okr.tenant-ids}") String[] tenantIds, Environment env) {
         this.env = env;
         for (String tenantId : tenantIds) {
             OauthConfig c = readOauthConfig(tenantId);
-            tenantConfigs.put(tenantId,
-                    createTenantConfig(c.jwkSetUri(), c.frontendClientIssuerUrl(), c.frontendClientId(), tenantId));
+            TenantConfig tenantConfig = createTenantConfig(c.jwkSetUri(), c.frontendClientIssuerUrl(),
+                    c.frontendClientId(), tenantId);
+
+            tenantConfigs.put(tenantId, tenantConfig);
+            cacheTenantConfig(tenantId, tenantConfig); // cache tenantConfig for Hibernate connections
         }
     }
 
+    private void cacheTenantConfig(String tenantId, TenantConfig tenantConfig) {
```

Comment:
Question: Do we really want to cache the config? I see the case that we want to do a demo of the project and therefore use multiple tenants, hence caching could disturb the flow. However i see great benefits in caching for the production environment.

Reply:
Since caching is tenant specific this should not oppose any problems

Reply:
Ja, möchten wir, aber das Caching ist immer pro Tenant.